### PR TITLE
feat: add support for event_trigger_retry

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes https://github.com/google-github-actions/deploy-cloud-functions/issues/183

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -154,9 +156,7 @@ async function run(): Promise<void> {
       labels: labels,
       maxInstances: maxInstances ? +maxInstances : undefined,
       minInstances: minInstances ? +minInstances : undefined,
-      // network: network, // TODO: add support
```
Comment (sethvargo) [inline src/main.ts:157]: network is deprecated

File: src/main.ts [hunk 2]
```diff
@@ -154,9 +156,7 @@ async function run(): Promise<void> {
...
       serviceAccountEmail: serviceAccountEmail,
-      // sourceToken: sourceToken, // TODO: add support
```
Comment (sethvargo) [inline src/main.ts:159]: sourceToken only applies to firebase

File: src/main.ts [hunk 3]
```diff
@@ -168,6 +168,12 @@ async function run(): Promise<void> {
         resource: eventTriggerResource,
         service: eventTriggerService,
       };
+
+      if (eventTriggerRetry) {
+        cf.eventTrigger.failurePolicy = {
+          retry: {},
```
Comment (bharathkkb) [inline src/main.ts:174]: Don't we need to specify something here if `eventTriggerRetry`?

Comment (sethvargo) [reply src/main.ts:174]: No, and it's dumb. `retry` is a `oneOf` with zero values. This is how you "enable" it. The default behavior is to omit the `failurePolicy` entirely (and thus not enable retries).

Comment (bharathkkb) [reply src/main.ts:174]: Ah gotcha, maybe lets add a comment explaining this? LGTM otherwise!

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**